uniref-rest/src/main/java/org/uniprot/api/uniref/queue/UniRefMessageListener.java

```diff
@@ -71,6 +77,14 @@ protected StoreRequest getStoreRequest(DownloadRequest request) {
         return StoreRequest.builder().addLineage(false).build();
     }
 
+    private Long getSolrHits(UniRefDownloadRequest request) {
+        UniRefSearchRequest searchRequest = new UniRefSearchRequest();
+        searchRequest.setQuery(request.getQuery());
+        searchRequest.setSize(0);
+        QueryResult<UniRefEntryLight> searchResults = service.search(searchRequest);
+        return searchResults.getPage().getTotalElements();
+    }
+
     @Override
```

Comment:
We can move getSolrHits to higher class with generic. I can see duplicate code in kb and uniref listener method getSolrHits.